src/lib.rs

```diff
@@ -993,133 +1162,53 @@ impl<L: SynthLanguage> Synthesizer<L> {
 
     /// Apply rewrites rules as they are being inferred, to minimize the candidate space.
     #[inline(never)]
-    fn choose_eqs(&mut self, mut new_eqs: EqualityMap<L>) -> (EqualityMap<L>, EqualityMap<L>) {
+    fn choose_eqs(
+        &mut self,
+        mut new_eqs: EqualityMap<L>,
+    ) -> (EqualityMap<L>, EqualityMap<L>) {
+        let step_sizes: Vec<usize> = vec![100, 10, 1];
         let mut bads = EqualityMap::default();
         let mut should_validate = true;
-        for step in vec![100, 10, 1] {
-            if self.params.rules_to_take < step {
-                continue;
-            }
-            let n_rules = usize::min(self.params.rules_to_take, new_eqs.len());
-            if step < 10 && n_rules > 200 {
-                break;
+        let mut step_idx = 0;
+
+        // Idea here is to remain at a high step level as long as possible
+        // Move down in step size if
+        //  (a) the number of eqs is smaller than the step size
+        //  (b) the number of bad eqs is smaller than the step size
```

Comment:
why b?

Reply:
I had a reason before, but I'm blanking on why that's the case. We can discuss this change.

Reply:
Ok yeah just explain to me sometime this week. I will resolve this for now.